Client/mods/deathmatch/logic/luadefs/CLuaWorldDefs.cpp

```diff
@@ -286,50 +289,70 @@ int CLuaWorldDefs::ProcessLineOfSight(lua_State* luaVM)
 
-                lua_pushnumber(luaVM, vecColNormal.fX);
-                lua_pushnumber(luaVM, vecColNormal.fY);
-                lua_pushnumber(luaVM, vecColNormal.fZ);
+                lua_pushnumber(L, vecColNormal.fX);
+                lua_pushnumber(L, vecColNormal.fY);
+                lua_pushnumber(L, vecColNormal.fZ);
 
-                lua_pushinteger(luaVM, iMaterial);
-                lua_pushnumber(luaVM, fLighting);
-                lua_pushinteger(luaVM, iPiece);
+                lua_pushinteger(L, iMaterial);
+                lua_pushnumber(L, fLighting);
+                lua_pushinteger(L, iPiece);
 
-                if (bIncludeBuildingInfo && buildingResult.bValid)
+                if (bIncludeBuildingInfo && buildingResult.bValid) // 8 args
                 {
-                    lua_pushnumber(luaVM, buildingResult.usModelID);
+                    lua_pushnumber(L, buildingResult.usModelID);
+
+                    lua_pushnumber(L, buildingResult.vecPosition.fX);
+                    lua_pushnumber(L, buildingResult.vecPosition.fY);
+                    lua_pushnumber(L, buildingResult.vecPosition.fZ);
+
+                    lua_pushnumber(L, ConvertRadiansToDegrees(buildingResult.vecRotation.fX));
+                    lua_pushnumber(L, ConvertRadiansToDegrees(buildingResult.vecRotation.fY));
+                    lua_pushnumber(L, ConvertRadiansToDegrees(buildingResult.vecRotation.fZ));
+
+                    lua_pushnumber(L, buildingResult.usLODModelID);
+                } else {
+                    for (auto i = 1 + 3 + 3 + 1; i-- > 0;) {
+                        lua_pushnil(L);
+                    }
+                }
 
-                    lua_pushnumber(luaVM, buildingResult.vecPosition.fX);
-                    lua_pushnumber(luaVM, buildingResult.vecPosition.fY);
-                    lua_pushnumber(luaVM, buildingResult.vecPosition.fZ);
+                if (bIncludeExtraMateriaInfo && matInfo.valid)  { // 7 args
+                    lua::Push(L, matInfo.uv.fX);
+                    lua::Push(L, matInfo.uv.fY);
 
-                    lua_pushnumber(luaVM, ConvertRadiansToDegrees(buildingResult.vecRotation.fX));
-                    lua_pushnumber(luaVM, ConvertRadiansToDegrees(buildingResult.vecRotation.fY));
-                    lua_pushnumber(luaVM, ConvertRadiansToDegrees(buildingResult.vecRotation.fZ));
+                    lua::Push(L, matInfo.textureName);
+                    lua::Push(L, matInfo.frameName);
 
-                    lua_pushnumber(luaVM, buildingResult.usLODModelID);
-                    return 19;
+                    lua::Push(L, matInfo.hitPos.fX);
+                    lua::Push(L, matInfo.hitPos.fY);
+                    lua::Push(L, matInfo.hitPos.fZ);
+                } else {
+                    for (auto i = 2 + 1 + 1 + 3; i-- > 0;) {
+                        lua_pushnil(L);
+                    }
                 }
-                return 11;
+
+                return 11 + 8 + 7;
```

Comment:
Maybe use variables instead? I don't expect anyone to change this code in the foreseeable time, but this seems odd.

Reply:
For the number of args?